src/main/java/it/gov/pagopa/payhub/pdnd/connector/pdnd/service/PdndClientAssertionBuilderService.java

```diff
@@ -0,0 +1,68 @@
+package it.gov.pagopa.payhub.pdnd.connector.pdnd.service;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JOSEObjectType;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSSigner;
+import com.nimbusds.jose.crypto.RSASSASigner;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import it.gov.pagopa.payhub.pdnd.config.pdnd.PdndConfig;
+import it.gov.pagopa.payhub.pdnd.config.pdnd.PdndBaseServiceIntegratedConfig;
+import it.gov.pagopa.payhub.pdnd.config.pdnd.PdndServiceIntegrationConfig;
+import it.gov.pagopa.payhub.pdnd.utils.CertUtils;
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.Date;
+import java.util.UUID;
+import org.springframework.stereotype.Service;
+
+@Service
+public class PdndClientAssertionBuilderService {
+
+  private final PdndConfig pdndConfig;
+  private final PdndBaseServiceIntegratedConfig pdndBaseServiceIntegratedConfig;
+
+  public PdndClientAssertionBuilderService(PdndConfig pdndConfig,
+      PdndBaseServiceIntegratedConfig pdndBaseServiceIntegratedConfig) {
+    this.pdndConfig = pdndConfig;
+    this.pdndBaseServiceIntegratedConfig = pdndBaseServiceIntegratedConfig;
+  }
+
+  public String buildPdndClientAssertion(PdndBaseServiceIntegratedConfig pdndBaseServiceIntegratedConfig,
+      PdndServiceIntegrationConfig pdndServiceIntegrationConfig)
+      throws InvalidKeySpecException, NoSuchAlgorithmException, IOException, JOSEException {
```

Comment:
These exceptions are an implementation details regarding how we are building the clientAssertion.
Instead of let them to be thrown, let's catch these exceptions and throws an unchecked exception (we are not interested on handling it, moreover these are related to deployment errors... let's throw just an IllegalStateException) (who will call this method will not know how to handle these exception anyway)